File: Dockerfile.go
FROM golang:1.22-alpine AS build
# Build stage: compile the Go source into a static binary
WORKDIR /src
COPY go/script.go .
RUN go build -o /out/app ./script.go

FROM alpine:3.20
# Runtime stage: only the compiled binary, small final image
WORKDIR /app
COPY --from=build /out/app ./app
CMD ["./app"]

File: Dockerfile.java
FROM eclipse-temurin:21-jdk AS build
# Build stage: compile Java source to bytecode
WORKDIR /src
COPY java/Hello.java .
RUN javac Hello.java

FROM eclipse-temurin:21-jre
# Runtime stage: runs with smaller JRE image
WORKDIR /app
COPY --from=build /src/Hello.class ./Hello.class
CMD ["java", "Hello"]
